src/rules/lambda/LambdaLatestVersion.ts

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+import { parse } from 'path';
+import { CfnFunction, Runtime } from '@aws-cdk/aws-lambda';
+import { CfnResource } from '@aws-cdk/core';
+import { NagRuleCompliance, NagRules } from '../../nag-rules';
+
+/**
+ * Lambda functions are configured to use the latest runtime version
+ * @param node the CfnResource to check
+ */
+export default Object.defineProperty(
+  (node: CfnResource): NagRuleCompliance => {
+    if (node instanceof CfnFunction) {
+      const runtime = NagRules.resolveIfPrimitive(node, node.runtime);
+      if (!runtime) {
+        // Runtime is not required for container lambdas, in this case, not applicable
+        return NagRuleCompliance.NOT_APPLICABLE;
+      }
+
+      const exp = /([a-z]+)(\d+(\.?\d+|\.x)?)?/;
+      const m = runtime.match(exp);
+
+      if (!m) {
+        // shouldn't happen, but if for some reason it does, we'll ignore the check
+        // CloudFormation will likely fail when trying to create the lambda
+        return NagRuleCompliance.NOT_APPLICABLE;
+      }
```

Comment:
Rather than saying this is Not Applicable (since that means this rule won't appear in reports for a particular Lambda function). I'd rather throw an error here and have it bubble up to be a `CdkNagValidationFailure` so the user can debug further if they choose. Tests (and possible how we run them) will also need to be adjusted for this. [Example](https://github.com/cdklabs/cdk-nag/blob/e081e7dfd11236f415311631a108028c0a1250c5/test/Engine.test.ts#L837)

```suggestion
      if (!m) {
        throw Error(
          `The Lambda runtime "${runtime}" does not match the expected regular expression, therefore the rule could not be validated.`
        )
      }
```